### PR TITLE
Feat: create Common/Pagination component

### LINKED ISSUE #5915: Create `Pagination` Component
Create a `Pagination` component using the new [figma design (direct link)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=495-11573&mode=dev). 

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).
- Review the [component creation guide](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-components) to follow our standards.
- ❗ Review the [translation guide](https://github.com/nodejs/nodejs.org/blob/main/TRANSLATION.md), especially how to [add new keys](https://github.com/nodejs/nodejs.org/blob/main/TRANSLATION.md#adding-new-translation-keys).

---

### Details

image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)
🌐 This story requires some internationalization support. Slow down!

- Create a new directory called `Pagination` within the [`Common` directory](https://github.com/nodejs/nodejs.org/tree/main/components/common) that contains all the new code. (This location may change in the future.)
- The `@heroicons/react` module has the icons `arrow-right` and `arrow-left` available for use
- Add a storybook story file `components/common/Pagination/index.stories.tsx` which exercises each of the component's states.
- `Previous` and `Next` must be internationalized. Add new keys `components.common.pagination.prev` and `components.common.pagination.next` to the `i18n/locales/en.json` file. `FormattedMessage` from `react-intl` should be used to reference the keys. Look for examples elsewhere in the codebase.
- always display pages 1, 2,3 and the last three, deduplicated
- style the current page according to the figma
- display an ellipsis when the total number of pages reaches 8 or more

image

- use responsive styles to wrap the previous and next buttons below the page links 

image

Suggested props include:

- `currentPage` (integer)
- `pages`, an array of objects that looks like this:

```js
[
    {
        value: "1",
        url: "blog/en/1"
    },
    {
        value: "2",
        url: "blog/en/2"
    },
    {
        value: "3",
        url: "blog/en/3"
    },
]
```

> urls subject to change as our blog pagination development continues

There are 3 collection states to capture with stories:

- less than 8 pages, current page visible
- 8 or more pages, current page visible
- 8 or more pages, current page in center with ellipsis on both sides

There are 4 layout states to capture within styles and stories:

- Light, Desktop
- Light, Mobile
- Dark, Desktop
- Dark, Mobile

### Open Questions

If the current page is not in the displayed list per our previous rules, how should it get displayed? Research other pagination patterns to determine how this should look.

Material UI would suggest something like this:

image

GitHub does something similar

image

### PR DESCRIPTION
**Pull request by danielmarcano**
## Description

- Create the `Common/Pagination` component, based on these [Figma designs](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=495-11573&mode=design&t=6oL4c6PhWGD9NW3C-4)
- It takes the following properties:

```ts
type Page = {
  url: string;
};

export type PaginationProps = {
  /**
   * One-based number of the current page
   */
  currentPage: number;
  pages: Page[];
  /**
   * The number of page buttons on each side of the current page button
   * @default 1
   */
  currentPageSiblingsCount?: number;
};
```

- It displays pages following a set of rules, all documented within the unit tests:
 - When there's 6 pages or less, all the page buttons are visible
 - When there's more than the minimum amount of visible pages, and:
   - There's at least two elements between the current page's left sibling and the first page, a left ellipsis is shown
   - There's at least two elements between the current page's right sibling and the last page, a right ellipsis is shown
   - When both of the previous points are true, two ellipses are shown on either side
   - Ellipses, when shown, appear between the first or last element, and the siblings of the current element

- It disables the previous button when the current page is the first one

- It disables the last button when the current page is the last one

- Proper accessibility ARIA attributes have been used, in order to make the component as accessible as possible

## Validation

- Here's a short video showcasing the stories for this new component

https://github.com/nodejs/nodejs.org/assets/93147810/9b037caf-3a67-408c-b160-b86d5bbd3b07

- Unit tests have been added to prevent regression errors on the entire rendering logic and behavior of the component

## Related Issues

Closes [#5915](https://github.com/nodejs/nodejs.org/issues/5915)

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Common/Pagination/__tests__/index.test.mjs
Comment (ovflowd) [inline components/Common/Pagination/__tests__/index.test.mjs:1]: Really good coverage. My only nitpick would be: Instead of calling the function `setUpTest`, it should probably be `renderPagination`...

Question: do these unit tests give you any warning? Such as `act` not being called? (Just out of curiosity)

File: components/Common/Pagination/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/index.module.css]: Interesting, I wonder why the choice of these complex area names... Couldn't you simplify it?

File: components/Common/Pagination/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/index.module.css]: This one got left out

File: components/Common/Pagination/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/index.module.css]: I'm unfamiliar with the negate statement here. What is it supposed to do 👀

File: components/Common/Pagination/index.stories.tsx
Comment (ovflowd) [inline components/Common/Pagination/index.stories.tsx]:
```suggestion
      {  url '1' },
```

All these multiline objects could become one line as they're too small. (Instead of occupying 3 lines per entry, just occupying one)

Comment (ovflowd) [reply components/Common/Pagination/index.stories.tsx]: (Suggestion for the whole file)

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: I don't think we should add `'use client';` at this point of time

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: Could you use our `LocalizedLink` instead of `next/link`?

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: These two objects probably can also be declared in one line, no?

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: We avoid using `return` statements when not needed within methods such as `.map`

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: Since intl is called just for one entry, I wonder if you couldn't just use `useIntl` (or withIntl) with dependency injection as passing `intl` as a prop (which feels odd)

Even tho this is a "function" React will pretty much treat this as a Functional Component.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: We use `//` doc blocks as comments.

File: components/Common/Pagination/index.tsx
Comment (ovflowd) [inline components/Common/Pagination/index.tsx:34]: I wonder how accessibility comes in play here. Can you navigate this Pagination just with ... keyboard strokes?

File: components/Common/Pagination/index.tsx
Comment (ovflowd) [inline components/Common/Pagination/index.tsx]: Is a span needed here? Afaik you can directly on `FormattedMessage` specify it to use a "span" element. So no need to surround with yet another span. I think that is even the default behavior but you can pass `tagName="span"`

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: This... should probably be a Component on its own.

Also, this Hook is sorta overcomplicated

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx:35]: I worry this whole function has little to none inline documentation of what each thing does.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [inline components/Common/Pagination/useGetPageElements.tsx]: I see a lot of duplicated logic / very similar logic. I wonder if recursivity could be used here or some sort of design pattern to simplify the logic and reduce the overall gigantic size of this Hook.

There are many moving parts here and little to none documentation. Like why "3 + 2 * something"... From where is that even coming from?

File: components/Common/Pagination/__tests__/index.test.mjs
Comment (danielmarcano) [reply components/Common/Pagination/__tests__/index.test.mjs:1]: Thank you! `renderPagination` sounds good, I will update the name of the function.

As for the warnings, no, there's no `act` warnings being thrown.

File: components/Common/Pagination/index.module.css
Comment (danielmarcano) [reply components/Common/Pagination/index.module.css]: Just wanted to be as informative as possible. What do you think of `pages`, `prev` and `next` as the new `grid-area` names?

File: components/Common/Pagination/index.module.css
Comment (danielmarcano) [reply components/Common/Pagination/index.module.css]: Got it, thank you.

File: components/Common/Pagination/index.module.css
Comment (danielmarcano) [reply components/Common/Pagination/index.module.css]: It adds `!important` to the declaration, which is something I avoid at all costs. Nonetheless, the thing is that anchor links are being applied some styles in the `base.css` file, and I thought that modifying those could have a negative impact in other areas that are counting with those base styles, so preferred to use `!` instead of modifying those:

image

File: components/Common/Pagination/index.stories.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.stories.tsx]: Got it, I will apply those changes

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Okay, thank you. I will remove these

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Yes, sure! I will replace them

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: By this you mean that they can be left like this?

```ts
aria-label={intl.formatMessage({ id: 'components.common.pagination.pageLabel' }, { pageNumber })}
```

If so, yes, I can change it.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: You mean the `return` statement on line `35`? If so, yes, I can remove it

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Using `useIntl` directly within the function was not an option, as that would prevent me from using this function conditionally, due to the rules of hooks.

Could you explain how would you do it with dependency injection? Isn't that what the `intl` prop is doing already?

Also, using `withIntil` for a regular function did not come to my mind. I can try that.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Okay, thank you. Will update this

File: components/Common/Pagination/index.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.tsx:34]: You can navigate it with tabs. There's no strong guidelines nor an official WCAG 2.1 recipe for Pagination components. However, there's tons of resources mentioning some of the core characteristics they can have:

- It should wrap everything through a nav or div with role="navigation"
- The pagination buttons should be wrapped in an ordered list, to let screen readers know the elements are ordered
- The list items should use `aria-setsize` and `aria-posinset`, as most of the times not all list items will be shown at once, so as to let screen readers know what the entire list size is, and the position of each list item in such list
- The list items should contain anchor links
- The attribute aria-current="page" should be used to indicate that such anchor link represents the current page

React MUI's [pagination component](https://mui.com/material-ui/react-pagination/) works with tabs too.

File: components/Common/Pagination/index.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.tsx]: I did not know that `FormattedMessage` added a span already. I will definitely remove the extra `span` element, then. Thank you

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: I agree with you in that it could be a component on its own. A sub-component called `PaginationListItem`, for example, with the following markup:

```tsx
  <li
        key={pageNumber}
        aria-setsize={parsedPages.length}
        aria-posinset={pageNumber}
      >
        <Link
          prefetch={false}
          href={url}
          aria-label={intl.formatMessage(
            {
              id: 'components.common.pagination.pageLabel',
            },
            {
              pageNumber,
            }
          )}
          className={styles.listItem}
          {...(pageNumber === currentPage && { 'aria-current': 'page' })}
        >
          <span>{pageNumber}</span>
        </Link>
      </li>

```

However, as per the complexity of the hook, the thing is that it's inherently a complex logic, although I have tried my best through the tests and the code to make it as explicative as possible.

I will replace the magic numbers 3 and 2 and add more comments, overall. I personally avoid large comment blocks, but I guess that in this case they would be helpful.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx:35]: Thank you. As mentioned in this [other comment](https://github.com/nodejs/nodejs.org/pull/5971#discussion_r1349698931), I will try to add comments that explain the different logical steps of the code, as I thought the code was pretty self-explicative but I now see that that's not the case.

The logic of this component is inherently complex, though, so the code will be a bit complex to understand but comments can definitely help.

You can take a look at React MUI [Pagination component's code](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/usePagination/usePagination.js) from the material-ui repository, which I took as inspiration. In my opinion, that code is a bit more difficult to follow.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Thank you. As mentioned in [this comment](https://github.com/nodejs/nodejs.org/pull/5971#discussion_r1349698931), the logic is inherently complex, and so the code, even with comments and after replacing the `3` and `2` magic numbers, will still be naturally "complex."

I thought about recursion but it would not make sense here, as we don't need to traverse the elements of the array. We just need to make calculations based on the array's size, and other variables at play: `currentPageSiblingsCount`, and `currentPage`'s values.

I will modify the code, remove magic numbers, add more comments to document what each step is doing, and try to reduce the amount of lines the custom hook has.

I agree with you in that it's complex, though, so will try my best to make it all simpler.

File: components/Common/Pagination/index.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.tsx]: I have checked, and `FormattedMessage` seems to just add the text by default, so no extra `span` was being added here.

image

It only enclosed the translation string within a tag when adding the `tagName` prop, so I can remove `span` and just pass it through the `tagName` property if that sounds better.

File: components/Common/Pagination/index.module.css
Comment (ovflowd) [reply components/Common/Pagination/index.module.css]: Sounds good to me

File: components/Common/Pagination/index.module.css
Comment (ovflowd) [reply components/Common/Pagination/index.module.css]: You probably want to do:

```css
.listItem a {
  @apply text-neutral-800
}
```

Probably instead of `!important`

File: components/Common/Pagination/index.tsx
Comment (ovflowd) [reply components/Common/Pagination/index.tsx:34]: Right, you did your due diligence. This looks good, and I'm satisfied with your explanation :)

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx]: Right, also I believe this should indeed be a sub-component. Can you do that?

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx]: Yup, `withIntl` should work. As again, for all intends and purposes this is a component, but I would rather that you extract this to its own file and have its own tests separated.

**Comment by danielmarcano** [comment]
Hi, @ovflowd, thank you for your comments and suggestions.

I have just pushed a refactor.

I have made the following changes:

- Moved markup and styles of page list items into their own `PaginationListItem` sub-component

- Did the same for `Ellipsis`

- Removed magic numbers and enhanced the logic of the `useGetPageElements` custom hook

- Documented the entire logic with comments, and improved the variable names

- Updated the `grid-area` names to single words: `pages`, `prev` and `next`

- Removed `span` and added `tagName` prop to `FormattedMessage` so that it rendered the `span` on its own

- Inlined all objects that Prettier formatting allowed me to

- Removed 'use client';

Please let me know if we would need anything else.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx]: Yup

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx]: No, sorry, I meant instead of:

```tsx
{
  pageNumber
}
```

it can be

```tsx
{ pageNumber }
```

(same for the id one)

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx:35]: > You can take a look at React MUI Pagination component's code from the material-ui repository, which I took as inspiration. In my opinion, that code is a bit more difficult to follow.

If that was used as inspiration, please add an inline comment mentioning that 🙇

File: components/Common/Pagination/useGetPageElements.tsx
Comment (ovflowd) [reply components/Common/Pagination/useGetPageElements.tsx]: > Thank you. As mentioned in this comment, the logic is inherently complex, and so the code, even with comments and after replacing the 3 and 2 magic numbers, will still be naturally "complex."

Right, understandable. Don't forget to make these magic numbers actual constants. So it's easier to update their usages, etc.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx:35]: Good catch!

I will add a comment referencing the file in an upcoming commit. Thank you!

**Comment by ovflowd** [comment]
> Hi, @ovflowd, thank you for your comments and suggestions.
> 
> I have just pushed a refactor.
> 
> I have made the following changes:
> 
> * Moved markup and styles of page list items into their own `PaginationListItem` sub-component
> * Did the same for `Ellipsis`
> * Removed magic numbers and enhanced the logic of the `useGetPageElements` custom hook
> * Documented the entire logic with comments, and improved the variable names
> * Updated the `grid-area` names to single words: `pages`, `prev` and `next`
> * Removed `span` and added `tagName` prop to `FormattedMessage` so that it rendered the `span` on its own
> * Inlined all objects that Prettier formatting allowed me to
> * Removed 'use client';
> 
> Please let me know if we would need anything else.

We super appreciate all the effort put in here. Thank you!

File: components/Common/Pagination/PaginationListItem/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/PaginationListItem/index.module.css]:
```suggestion
    h-10
```

File: components/Common/Pagination/PaginationListItem/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/PaginationListItem/index.module.css]: Don't forget the `.listeItem a` thingy that I've mentioned here.

File: components/Common/Pagination/PaginationListItem/index.tsx
Comment (ovflowd) [inline components/Common/Pagination/PaginationListItem/index.tsx]: Friendly reminder that we use the `//` for inline documentation

File: components/Common/Pagination/__tests__/index.test.mjs
Comment (ovflowd) [inline components/Common/Pagination/__tests__/index.test.mjs]: Can we add this as a sort of global mock? I bet somewhere this is already being mocked, if not, probably suits making a global mock 👀

File: components/Common/Pagination/index.tsx
Comment (ovflowd) [inline components/Common/Pagination/index.tsx]:
```suggestion
type Page = { urL: string };`
```

File: components/Common/Pagination/index.tsx
Comment (ovflowd) [inline components/Common/Pagination/index.tsx]: In general style imports should be the last thing imported. What happens if you change this to this:

```tsx
import { useGetPageElements } from './useGetPageElements';

import styles from './index.module.css';
```

File: components/Common/Pagination/index.module.css
Comment (danielmarcano) [reply components/Common/Pagination/index.module.css]: This did not work. The thing is that `.listItem` is actually being set on the anchor already, so unless we modify those base styles, we will need to keep important.

File: components/Common/Pagination/useGetPageElements.tsx
Comment (danielmarcano) [reply components/Common/Pagination/useGetPageElements.tsx]: Ohh, okay, great. I will add it this change in an upcoming commit. Thank you!

File: components/Common/Pagination/index.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.tsx:34]: Thank you ✨

File: components/Common/Pagination/index.tsx
Comment (danielmarcano) [reply components/Common/Pagination/index.tsx]: Awesome, thank you. I had to modify the path of the `useGetPageElements` import for this to work, as `ESLint` was the one formatting that import statement and placing it below the `styles` one.

File: components/Common/Pagination/PaginationListItem/index.module.css
Comment (danielmarcano) [reply components/Common/Pagination/PaginationListItem/index.module.css]: It did not work, as I had already set `.listItem` to the anchor, but that was not working... What ended up working was to target the same pseudo-classes as in the base.css styles:

```css
.listItem,
.listItem:link,
.listItem:active {
  @apply aria-current:bg-green-600
    aria-current:text-white
    flex
    h-10
    w-10
    items-center
    justify-center
    rounded
    px-3
    py-2.5
    text-neutral-800
    hover:bg-neutral-100
    dark:text-neutral-200
    hover:dark:bg-neutral-900;
}

```

File: components/Common/Pagination/__tests__/index.test.mjs
Comment (danielmarcano) [reply components/Common/Pagination/__tests__/index.test.mjs]: That's a great idea. I have created a `nextJsRouter.mjs` mock within a `__mocks__` directory, and imported it within `jest.setup.mjs`. I have also removed the existing `next/router` mocks in test files, and made sure the tests are still successful.

**Comment by danielmarcano** [comment]
Hi, @ovflowd. I have pushed all the latest changes and refactors you mentioned, including the removal of the `!important` rules, some more inlined objects, unit tests for `PaginationListItem`, and the global `nextJsRouter.mjs` mock so that we don't have to do so manually in each test file.

Let me know if the PR would be good to go, when you have some time.

Have a great rest of the day, and thanks for your kind feedback 🥳

File: __mocks__/nextJsRouter.mjs
Comment (ovflowd) [inline __mocks__/nextJsRouter.mjs]: Let's make `asPath: '/'`

File: components/Common/Pagination/PaginationListItem/index.module.css
Comment (ovflowd) [inline components/Common/Pagination/PaginationListItem/index.module.css:1]: Nice!

File: jest.setup.mjs
Comment (ovflowd) [inline jest.setup.mjs:2]: I think you can add the mock files on https://medium.com/@arivu_a/how-to-mock-functions-globally-in-jest-f267fedf7683 (setupFiles) instead of a manual import here

As in general, Jest should import this whenever it believes it should be imported (risk of this being imported when Jest Env is still not prepared)

**Review by ovflowd — APPROVED** [review]
Fantastic work! Thank you so much for your due diligence here. We really appreciate the effort put in here 🙇 

I've left some final comments, but besides that, I think we're good to go.

File: jest.setup.mjs
Comment (danielmarcano) [reply jest.setup.mjs:2]: I had not thought about this, thank you.

Perhaps `setupFilesAfterEnv` would be better, then, since those are run after the test framework is installed in the environment, as per the [docs](https://jestjs.io/docs/configuration#setupfilesafterenv-array)?

Edit: I have just checked, and `jest.setup.mjs` is actually being passed to `setupFilesAfterEnv`, so I would say it's good as it is, as `setupFilesAfterEnv` makes sure that any setup file passed to it is always executed after jest env has been prepared.

**Comment by danielmarcano** [comment]
Thank you, @ovflowd! I have modified `nextJsRouter.mjs` with the small change you mentioned. I have also left you an explanation on why I think `setupFilesAfterEnv` already does the job. Let me know if we would be good to go.

Also, I have a couple of final doubts, as this is my first contribution to this project:

- Are PRs merged by collaborators who have reviewed it, or by the author of the PR once it's been approved and all comments have been resolved?

- The `CONTRIBUTING.md` file mentions in the "When merging" section that "[squash]" is to be used in PRs made up of multiple commits. Is "squash and merge" already configured for each PR we create? Or should we manually squash the commits ourselves before merging the PR?

- There's 4 required checks in this PR that are all showing the same _"Expected — Waiting for status to be reported"_ message. These are: "Build on ubuntu-latest," "Build on windows-latest," "Lint," and "Tests." When are those checks triggered? Are they automatically triggered or is there anything else from my side that I should do to trigger them?

Thank you!

**Comment by ovflowd** [comment]
> Are PRs merged by collaborators who have reviewed it, or by the author of the PR once it's been approved and all comments have been resolved?

By collaborators.

> The CONTRIBUTING.md file mentions in the "When merging" section that "[squash]" is to be used in PRs made up of multiple commits. Is "squash and merge" already configured for each PR we create? Or should we manually squash the commits ourselves before merging the PR?

Automatically done by GitHub

> There's 4 required checks in this PR that are all showing the same "Expected — Waiting for status to be reported" message. These are: "Build on ubuntu-latest," "Build on windows-latest," "Lint," and "Tests." When are those checks triggered? Are they automatically triggered or is there anything else from my side that I should do to trigger them?

The checks are not triggered automatically, but rather by collaborators, to avoid unsafe Actions from running.

**Comment by danielmarcano** [comment]
Awesome, then, @ovflowd! If you think we are good to go, I leave the rest in your hands. Thanks for everything 🥳

**Pull request merged**
